File: OLSource1/v1/VS_officedev/codesnippet/CSharp/programming-vsto-add-ins_1.cs
        private void ThisAddIn_Startup(object sender, System.EventArgs e)
        {
            this.Application.DocumentOpen +=
    new Word.ApplicationEvents4_DocumentOpenEventHandler(WorkWithDocument);

            ((Word.ApplicationEvents4_Event)this.Application).NewDocument +=
                new Word.ApplicationEvents4_NewDocumentEventHandler(WorkWithDocument);  
        }
        
        private void WorkWithDocument(Microsoft.Office.Interop.Word.Document Doc)
        {
            try
            {
                Word.Range rng = Doc.Range(0, 0);
                rng.Text = "New Text";
                rng.Select();
            }
            catch (Exception ex)
            {
                // Handle exception if for some reason the document is not available.
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/programming-vsto-add-ins_1.vb
    Private Sub ThisAddIn_Startup() Handles Me.Startup

        AddHandler Application.NewDocument, AddressOf WorkWithDocument

    End Sub

    Private Sub WorkWithDocument(ByVal Doc As Microsoft.Office.Interop.Word.Document) _
        Handles Application.DocumentOpen

        Dim rng As Word.Range = Doc.Range(Start:=0, End:=0)
        rng.Text = " New Text "
        rng.Select()

    End Sub